xds/internal/balancer/ringhash/e2e/ringhash_balancer_test.go

```diff
@@ -2847,11 +2861,16 @@ func (s) TestRingHash_RequestHashKeyConnecting(t *testing.T) {
 	}()
 	testutils.AwaitState(ctx, t, cc, connectivity.Connecting)
```

Comment:
Pickfirst will report `CONNECTING` as soon as it is created. It doesn't wait for a subchannel to report `CONNECTING`. That's the reason this check wasn't sufficient. Should we remove it? The check for 1 connection attempt below seems to be sufficient.